From 4e3605d5bee5e06c2cc800f3ad6087208922177b Mon Sep 17 00:00:00 2001
From: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Date: Thu, 18 Mar 2021 10:19:41 -0700
Subject: [PATCH 735/767] Revert " octeontx2-af: Reset MSIX config space in
 FLR"

commit 6ad9f70a8906154c571de1adfdb0d22bc6eafbcc from
git@git.assembla.com:cavium/WindRiver.linux.git

This reverts commit 14366ff9312294487b17fcb2135b7289d3d22fa8.

PF_RST on a FLR is causing RVU_PF_VF_BAR4_ADDR also getting cleared. Due to that RVU_PF_VF_BAR4_ADDR is also getting cleared which would break PF::VF communication

Change-Id: Ifc1609a9b37144f039f11e888ba276957effb3ab
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/48051
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 .../net/ethernet/marvell/octeontx2/af/rvu.c   | 26 -------------------
 .../ethernet/marvell/octeontx2/nic/otx2_pf.c  |  6 -----
 2 files changed, 32 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
index 1a2a6583420d..10e9ead10051 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
@@ -2580,26 +2580,6 @@ static void rvu_sso_pfvf_rst(struct rvu *rvu, u16 pcifunc)
 	}
 }
 
-/* Reset PF/VF MSIX config space */
-static void rvu_pfvf_msix_reset(struct rvu *rvu, u16 pcifunc)
-{
-	u64 reg, val;
-	int pfvf;
-
-	if (pcifunc & RVU_PFVF_FUNC_MASK) {
-		pfvf = rvu_get_hwvf(rvu, pcifunc);
-		reg = RVU_AF_HWVF_RST;
-		val = (pfvf & 0xFF) | BIT_ULL(12);
-	} else {
-		pfvf = rvu_get_pf(pcifunc);
-		reg = RVU_AF_PF_RST;
-		val = (pfvf & 0x1F) | BIT_ULL(12);
-	}
-
-	rvu_write64(rvu, BLKADDR_RVUM, reg, val);
-	rvu_poll_reg(rvu, BLKADDR_RVUM, reg, BIT_ULL(12), true);
-}
-
 static void __rvu_flr_handler(struct rvu *rvu, u16 pcifunc)
 {
 	mutex_lock(&rvu->flr_lock);
@@ -2620,7 +2600,6 @@ static void __rvu_flr_handler(struct rvu *rvu, u16 pcifunc)
 	rvu_blklf_teardown(rvu, pcifunc, BLKADDR_NPA);
 	rvu_detach_rsrcs(rvu, NULL, pcifunc);
 	rvu_sso_pfvf_rst(rvu, pcifunc);
-	rvu_pfvf_msix_reset(rvu, pcifunc);
 	mutex_unlock(&rvu->flr_lock);
 }
 
@@ -2639,11 +2618,6 @@ static void rvu_afvf_flr_handler(struct rvu *rvu, int vf)
 	/* Signal FLR finish and enable IRQ */
 	rvupf_write64(rvu, RVU_PF_VFTRPENDX(reg), BIT_ULL(vf));
 	rvupf_write64(rvu, RVU_PF_VFFLR_INT_ENA_W1SX(reg), BIT_ULL(vf));
-	/* Re-enable MBOX and ME interrupt as it gets cleared
-	 * in HWVF_RST reset.
-	 */
-	rvupf_write64(rvu, RVU_PF_VFME_INT_ENA_W1SX(reg), BIT_ULL(vf));
-	rvupf_write64(rvu, RVU_PF_VFPF_MBOX_INT_ENA_W1SX(0), BIT_ULL(vf));
 }
 
 static void rvu_flr_handler(struct work_struct *work)
diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index 055b328b01c8..d506300a4123 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -125,12 +125,6 @@ static void otx2_flr_handler(struct work_struct *work)
 		/* clear transcation pending bit */
 		otx2_write64(pf, RVU_PF_VFTRPENDX(reg), BIT_ULL(vf));
 		otx2_write64(pf, RVU_PF_VFFLR_INT_ENA_W1SX(reg), BIT_ULL(vf));
-		/* Re-enable MBOX and ME interrupt as it gets cleared
-		 * in HWVF_RST reset.
-		 */
-		otx2_write64(pf, RVU_PF_VFME_INT_ENA_W1SX(reg), BIT_ULL(vf));
-		otx2_write64(pf, RVU_PF_VFPF_MBOX_INT_ENA_W1SX(reg),
-			     BIT_ULL(vf));
 	}
 
 	mutex_unlock(&mbox->lock);
-- 
2.31.1

